pkg/descheduler/evictions/evictions.go

```diff
@@ -97,19 +98,21 @@ func (pe *PodEvictor) TotalEvicted() int {
 // EvictPod returns non-nil error only when evicting a pod on a node is not
 // possible (due to maxPodsToEvictPerNode constraint). Success is true when the pod
 // is evicted on the server side.
-func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node, reasons ...string) (bool, error) {
+func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node, strategy string, reasons ...string) (bool, error) {
```

Comment:
Is this new `strategy` param used anywhere? It's kind of confusing to have to update all the strategies with lines like `podEvictor.EvictPod(ctx, pod, node, "NodeAffinity", "NodeAffinity")`

Reply:
I don't like it either. We might drop `reasons` field since I don't see it used for anything but passing a strategy name.

Reply:
I think `reasons` is fine, but specifically the `strategy` param you added here... I don't see it used anywhere, and it seems like you're just using `reasons` anyway, right?

Reply:
Oh, I see. My bad. `metrics.PodsEvicted.With(map[string]string{"result": "maximum number reached", "strategy": reason, "namespace": pod.Namespace}).Inc()` is supposed to have `"strategy": strategy` pair in it.

Reply:
Ah yeah, I agree then we could either drop the `...reasons` param all together, *or* add this strategy param, and just use that in each strategy (since reasons is optional, the strategy can just pass its name which will be used for metrics).

I can see a possible use case where one strategy may have different reasons for eviction, so I'm not sure it's worth getting rid of entirely yet. But it doesn't make much sense to keep if we're not using it. Wdyt?

Reply:
> I'm not sure it's worth getting rid of entirely yet

Yeah, not completely sure either. I'll keep it there for the moment. Better than drop it and re-introduce it later.